Mbed update

jenswi-linaro opened the pull request:
Update mbedtls with two commits to help integration in libutee

To make this branch compile we need to merge master into import/mbedtls-2.6.1 again, https://github.com/OP-TEE/optee_os/pull/2678 need to be part of that merge.

jforissier commented on lib/libmbedtls/mbedtls/library/bignum.c:
```diff
@@ -2177,7 +2177,7 @@ static int mpi_miller_rabin( const mbedtls_mpi *X,
             }
 
-            if (count++ > 30) {
+            if (count++ > 300) {
```
Why?

jenswi-linaro replied:
Without this this function sometimes (quite often) incorrectly dismisses the number 000000022770A7DC599BC90B2FF981CCB5CF05703344C8F350418AAD as not being a prime number.

jenswi-linaro replied:
I'll update the commit message.

jforissier replied:
Ha! Yet, 000000022770A7DC599BC90B2FF981CCB5CF05703344C8F350418AAD is soooooo obviously prime :) Seriously how did figure this out? Test suite?

jforissier commented on lib/libmbedtls/mbedtls/include/mbedtls/bignum.h:
```diff
@@ -196,6 +203,8 @@ extern void *mbedtls_mpi_mempool;
 void mbedtls_mpi_init( mbedtls_mpi *X );
 void mbedtls_mpi_init_mempool( mbedtls_mpi *X );
+void mbedtls_mpi_init_static( mbedtls_mpi *X , mbedtls_mpi_uint *p,
+                              int sign, size_t alloc_size, size_t nblimbs);
```
This needs a specific comment, especially since (as I understand it) the sentence "This just makes it ready to be set or freed, but does not define a value for the MPI" is not true here.

jforissier commented:
Candidates for upstream maybe? Anyway:

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

I have merged `master` into `import/mbedtls-2.6.1`, so if you rebase your branch it should build fine.

jenswi-linaro commented:
Rebased, squashed and applied tag.

Yes, at least the Miller-rabin commit should be a candidate for upstream.

jforissier commented:
Travis fails on checkpatch and more importantly on xtest 8101, can you take a look?

jenswi-linaro commented:
Updated commit "￼mbedtls: add mbedtls_mpi_init_static()"

Updated commit message and added this patch.

```patch
 void mbedtls_mpi_init_mempool( mbedtls_mpi *X )
 {
-    mpi_init(X, MBEDTLS_MPI_ALLOC_TYPE_MEMPOOL, NULL, 1, 0, 0);
+    if( mbedtls_mpi_mempool )
+        mpi_init(X, MBEDTLS_MPI_ALLOC_TYPE_MEMPOOL, NULL, 1, 0, 0);
+    else
+        mbedtls_mpi_init( X );
 }
```

jenswi-linaro commented:
Travis and Shippable are happy now.
